[Modular] update the auto pipeline blocks doc (#13148)
huggingface/diffusers: docs/source/en/modular_diffusers/auto_pipeline_blocks.md

        )
```

> It is **very** important to include a `description` to avoid any confusion over how to run a block and what inputs are required. While [`~modular_pipelines.AutoPipelineBlocks`] are convenient, its conditional logic may be difficult to figure out if it isn't properly explained.

Reviewer: Should we consider providing some kinda of templating in this case? DOesn't have to be in this PR, though.

## Workflows

Pipelines that contain conditional blocks ([~modular_pipelines.AutoPipelineBlocks] or [~modular_pipelines.ConditionalPipelineBlocks]) can support multiple workflows — for example, our SDXL modular pipeline supports a dozen workflows all in one pipeline. But this also means it can be confusing for users to know what workflows are supported and how to run them. For pipeline builders, it's useful to be able to extract only the blocks relevant to a specific workflow.

We recommend defining a `_workflow_map` to give each workflow a name and explicitly list the inputs it requires.

```py
from diffusers.modular_pipelines import SequentialPipelineBlocks

class MyPipelineBlocks(SequentialPipelineBlocks):
    block_classes = [TextEncoderBlock, AutoImageBlocks, DecodeBlock]
    block_names = ["text_encoder", "auto_image", "decode"]

    _workflow_map = {
        "text2image": {"prompt": True},
        "image2image": {"image": True, "prompt": True},
        "inpaint": {"mask": True, "image": True, "prompt": True},
    }
```

>  All of our built-in modular pipelines come with workflows defined. You can list them all supported workflows like this:

Reviewer 2: ```suggestion
All of our built-in modular pipelines come with pre-defined workflows. The `available_workflows` property lists all supported workflows:
```
